src/Arrays/vector_operations.jl

```diff
@@ -0,0 +1,241 @@
+Remove duplicate entries in a sorted vector.
+
+### Input
+
+- `v` -- sorted vector
+
+### Output
+
+The input vector without duplicates.
+"""
+function remove_duplicates_sorted!(v::AbstractVector)
+    for i in length(v)-1:-1:1
+        if v[i] == v[i+1]
+            splice!(v, i+1)
+        end
+    end
+    return v
+end
+
+"""
+    samedir(u::AbstractVector{N},
+            v::AbstractVector{N})::Tuple{Bool, Real} where {N<:Real}
+
+Check whether two vectors point in the same direction.
+
+### Input
+
+- `u` -- first vector
+- `v` -- second vector
+
+### Output
+
+`(true, k)` iff the vectors are identical up to a positive scaling factor `k`,
+and `(false, 0)` otherwise.
+
+
+### Examples
+
+```jldoctest
+julia> using LazySets: samedir
+
+julia> samedir([1, 2, 3], [2, 4, 6])
+(true, 0.5)
+
+julia> samedir([1, 2, 3], [3, 2, 1])
+(false, 0)
+
+julia> samedir([1, 2, 3], [-1, -2, -3])
+(false, 0)
+
+```
+"""
+function samedir(u::AbstractVector{N},
+                 v::AbstractVector{N}
+                )::Tuple{Bool, Real} where {N<:Real}
+    @assert length(u) == length(v) "wrong dimension"
+    no_factor = true
+    factor = 0
+    @inbounds for i in 1:length(u)
+        if u[i] == 0
+            if v[i] != 0
+                return (false, 0)
+            end
+            continue
+        elseif v[i] == 0
+            return (false, 0)
+        end
+        if no_factor
+            no_factor = false
+            factor = u[i] / v[i]
+            if factor < 0
+                return (false, 0)
+            end
+        elseif factor != u[i] / v[i]
+            return (false, 0)
+        end
+    end
+    if no_factor
+        # both vectors are zero
+        return (true, 0)
+    end
+    return (true, factor)
+end
+
+"""
+    nonzero_indices(v::AbstractVector{N})::Vector{Int} where {N<:Real}
+
+Return the indices in which a vector is non-zero.
+
+### Input
+
+- `v` -- vector
+
+### Output
+
+A vector of ascending indices `i` such that the vector is non-zero in dimension
+`i`.
+"""
+function nonzero_indices(v::AbstractVector{N})::Vector{Int} where {N<:Real}
+    n = length(v)
+    res = Vector{Int}()
+    sizehint!(res, n)
+    for i in 1:n
+        if v[i] != zero(N)
+            push!(res, i)
+        end
+    end
+    return res
+end
+
+function nonzero_indices(v::SparseVector{N})::Vector{Int} where {N<:Real}
+    return v.nzind
+end
+
+"""
+    rectify(x::AbstractVector{N}) where {N<:Real}
+
+Rectify a vector, i.e., take the element-wise maximum with zero.
+
+### Input
+
+- `x` -- vector
+
+### Output
+
+A copy of the vector where each negative entry is replaced by zero.
+"""
+function rectify(x::AbstractVector{N}) where {N<:Real}
+    return map(xi -> max(xi, zero(N)), x)
+end
+
+"""
+    iscounterclockwise(result::AbstractVector,
+                       correct_expr::AbstractVector) where {N<:Real}
+
+Returns a boolean, true if the elements in `result` are ordered in
+a couter-clockwise fashion and in the same order as `correct_expr`.
```

Comment:
```suggestion
a counter-clockwise fashion and in the same order as `correct_expr`.
```

Reply:
Since I copied this only and this is going to change in #1463, I rule that this stays unchanged :crown: